Mathlib/Algebra/Ring/Equiv.lean

```diff
@@ -833,16 +833,28 @@ end SemiringHom
 
 section GroupPower
 
+variable {F : Type*} {M : Type*} {N : Type*} [EquivLike F M N]
+
+protected
+theorem map_isUnit_iff {M N} [Monoid M] [Monoid N] [EquivLike F M N] [MonoidHomClass F M N]
+    (f : F) {m : M} : IsUnit m ↔ IsUnit (f m) := by
+  refine ⟨(IsUnit.map f ·), fun h ↦ ?_⟩
+  let g := MonoidHom.inverse (A := M) (B := N) f (EquivLike.inv f) (EquivLike.left_inv f)
+    (EquivLike.right_inv f)
+  exact (EquivLike.left_inv f m) ▸ IsUnit.map g h
```

Comment:
This presumably doesn't belong in this file any more, since it is not about rings.

Reply:
I agree, but as I said in a reply to your previous comment, I tried to put it inside `Group\Equiv\Basic.lean`, where I think it should belong, but this creates import loops.

Reply:
This one definitely shouldn't be in the RingEquiv namespace; MulEquiv could be acceptable. In analogy to [isUnit_map_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Ideal/LocalRing.html#isUnit_map_iff) (which is for local ring homs), we should swap the two sides and align the name:
```
theorem map_isUnit_iff {M N} [Monoid M] [Monoid N] [EquivLike F M N] [MonoidHomClass F M N]
    (f : F) {m : M} : IsUnit (f m) ↔ IsUnit m :=
  isUnit_map_of_leftInverse (MonoidHom.inverse (f : M →* N) (EquivLike.inv f)
    (EquivLike.left_inv f) <| EquivLike.right_inv f) (EquivLike.left_inv f)
```
#find_home says it can go into Mathlib.Algebra.Group.Units.Equiv. I think MulHomClass also works but we can't use `isUnit_map_of_leftInverse` with it.